core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -715,40 +743,58 @@ boolean computeIfPresent(K key, Consumer<OakWBuffer> computer) {
             throw new NullPointerException();
         }
 
-        Chunk c = findChunk(key); // find chunk matching key
+        Chunk<K, V> c = findChunk(key); // find chunk matching key
         Chunk.LookUp lookUp = c.lookUp(key);
-        if (lookUp == null || lookUp.handle == null) return false;
+        if (lookUp == null || lookUp.valueSlice == null) {
+            return false;
+        }
 
-        return lookUp.handle.compute(computer);
+        ValueUtils.ValueResult res = ValueUtils.compute(lookUp.valueSlice, computer);
+        if (res == RETRY) {
```

Comment:
I don't know if you are going to continue with "your result" (ValueResult) after merging with Eran's last code. But Eran had a class Result. There need to be one "result" in the code after all...

Reply:
My result has 3 values, `SUCCESS, FAILURE` and `RETRY` and Eran uses a boolean result.
I could try and consolidate them both into one.